app/src/main/java/com/github/onlynotesswent/ui/overview/EditNote.kt

```diff
@@ -36,62 +49,167 @@ import com.google.firebase.Timestamp
+                    Text("Edit note", Modifier.testTag("editNoteTitle"))
+                    Spacer(modifier = Modifier.weight(2f))
+                  }
+            },
+            navigationIcon = {
+              IconButton(
+                  onClick = { navigationActions.goBack() }, Modifier.testTag("goBackButton")) {
+                    Icon(
+                        imageVector = Icons.AutoMirrored.Outlined.ArrowBack,
+                        contentDescription = "Back")
+                  }
+            })
+      },
+      content = { paddingValues ->
+        LazyColumn(
+            modifier =
+                Modifier.fillMaxSize()
+                    .padding(16.dp)
+                    .padding(paddingValues)
+                    .testTag("editNoteColumn"),
+            verticalArrangement = Arrangement.spacedBy(8.dp),
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              item {
+                OutlinedTextField(
+                    value = updatedNoteTitle,
+                    onValueChange = { updatedNoteTitle = it },
+                    label = { Text("Note Title") },
+                    placeholder = { Text("Enter the new title here") },
+                    modifier = Modifier.fillMaxWidth().testTag("EditTitle textField"))
+              }
 
-        OutlinedTextField(
-            value = updatedNoteTitle,
-            onValueChange = { newTitle: String -> updatedNoteTitle = newTitle },
-            label = { Text("Note Title") },
-            placeholder = { Text("Enter the new title here") },
-            modifier = Modifier.fillMaxWidth().testTag("EditTitle textField"))
+              item {
+                OptionDropDownMenu(
+                    value = visibility,
+                    expanded = expandedVisibility,
+                    buttonTag = "visibilityEditButton",
+                    menuTag = "visibilityEditMenu",
+                    onExpandedChange = { expandedVisibility = it },
+                    items = listOf("Public", "Private"),
+                    onItemClick = { visibility = it })
+              }
 
-        OutlinedTextField(
-            value = updatedNoteText,
-            onValueChange = { newText: String -> updatedNoteText = newText },
-            label = { Text("Note Content") },
-            placeholder = { Text("Enter your note here...") },
-            modifier = Modifier.fillMaxWidth().height(400.dp).testTag("EditNote textField"))
+              item {
+                OutlinedTextField(
+                    value = updatedClassName,
+                    onValueChange = { updatedClassName = it },
+                    label = { Text("Class Name") },
+                    placeholder = { Text("Set the Class Name for the Note") },
+                    modifier = Modifier.fillMaxWidth().testTag("EditClassName textField"))
+              }
 
-        Button(
-            onClick = {
-              noteViewModel.updateNote(
```

Comment:
We should update the note only if there are no errors. For example like said previously, we should check the title is not null.  We should disable the button if the note is not null